bundles/azure/src/main/java/org/apache/gravitino/abs/fs/AzureFileSystemProvider.java

```diff
@@ -58,13 +68,39 @@ public FileSystem getFileSystem(@Nonnull Path path, @Nonnull Map<String, String>
           config.get(AzureProperties.GRAVITINO_AZURE_STORAGE_ACCOUNT_KEY));
     }
 
-    if (!config.containsKey(ABFS_IMPL_KEY)) {
+    if (!hadoopConfMap.containsKey(ABFS_IMPL_KEY)) {
       configuration.set(ABFS_IMPL_KEY, ABFS_IMPL);
     }
 
     hadoopConfMap.forEach(configuration::set);
 
-    return FileSystem.get(path.toUri(), configuration);
+    return FileSystem.newInstance(path.toUri(), configuration);
+  }
+
+  @Override
+  public Map<String, String> getFileSystemCredentialConf(Credential[] credentials) {
+    Credential credential = AzureStorageUtils.getSuitableCredential(credentials);
```

Comment:
you couldn't get `AzureAccountKeyCredential` from `getSuitableCredential` any more, use a new method?

Reply:
This should be in one method. I have revised the code.